### PR TITLE
Update build function to receive image name & version ⚡️

### PR DESCRIPTION
**Pull request by anuruddhal**
## Purpose

- Refactor build function
- Update CLI dependencies
- Update CLI to pass image name & version
- Copy swagger resources to target folder
- Update samples to new syntax

### CODE REVIEW

File: components/cli/pkg/util/utils.go [hunk 1]
```diff
@@ -532,7 +533,7 @@ func RenameFile(oldName, newName string) error {
 }
 
 func ReplaceFile(fileToBeReplaced, fileToReplace string) error {
-	errRename := RenameFile(fileToBeReplaced, fileToBeReplaced + "-old")
+	errRename := RenameFile(fileToBeReplaced, fileToBeReplaced+"-old")
```
Comment (sinthuja) [inline components/cli/pkg/util/utils.go]: This looks like a formatting issue right? Isn't it required to have the space after '+'?

File: components/cli/pkg/util/utils.go [hunk 2]
```diff
@@ -601,7 +602,7 @@ func ExtractTarGzFile(extractTo, archive_name string) error {
 	cmd := exec.Command("tar", "-zxvf", archive_name)
 	cmd.Dir = extractTo
 
-	ExecuteCommand(cmd, "Error occured in extracting file :" + archive_name)
+	ExecuteCommand(cmd, "Error occured in extracting file :"+archive_name)
```
Comment (sinthuja) [inline components/cli/pkg/util/utils.go]: Same as above.

File: components/lang/src/main/java/io/cellery/CelleryConstants.java [hunk 3]
```diff
@@ -32,4 +32,8 @@
     public static final String CELL_REFERENCE_TEMPLATE_CONTEXT_COMPONENTS = "components";
     public static final String CELL_REFERENCE_TEMPLATE_CONTEXT_HANDLE_API_NAME = "handleApiName";
     public static final String CELL_REFERENCE_TEMPLATE_CONTEXT_HANDLE_TYPE_NAME = "handleTypeName";
+
+    public static final String TARGET = "target";
+    public static final String RESOURCES = "resources";
+    public static final String GATEWAY_PROTOCOL = "http";
```
Comment (sinthuja) [inline components/lang/src/main/java/io/cellery/CelleryConstants.java]: As the gateway protocol can be different transports, any reason for this to stick to HTTP? We can probably rename this to HTTP_GATEWAY_PROTOCOL as the constant name, or if it's default protocol, let's change this to DEFAULT_GATEWAY_PROTOCOL

Comment (anuruddhal) [reply components/lang/src/main/java/io/cellery/CelleryConstants.java]: This is the default gateway protocol. Changed it to DEFAULT_GATEWAY_PROTOCOL. (https://github.com/cellery-io/sdk/pull/25/commits/427234b689041403a2033292514d323df5267697)

File: components/cli/pkg/util/utils.go (hunk 1 above)
Comment (anuruddhal) [reply components/cli/pkg/util/utils.go]: This is the go formatting standard.

File: components/cli/pkg/util/utils.go (hunk 2 above)
Comment (anuruddhal) [reply components/cli/pkg/util/utils.go]: This is the go formatting standard.

**Pull request merged**